### PR TITLE
Issue 1526 slack interceptors

### PR DESCRIPTION
**Pull request by ilan-pinto**
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [ ] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

**Comment by ilan-pinto** [comment]
@savitaashture @bigfleet it seems like the CI server has som resource
 issues not related to my PR. Could you check it

**Comment by ilan-pinto** [comment]
@savitaashture is ready to review

File: Makefile [hunk 1]
```diff
@@ -198,3 +198,14 @@ help:
...
+
+.PHONY:  deploy_tekton
+deploy_tekton:  | ; $(info $(M) deploying tekton on local cluster …) @ ## Deploying tekton on local clustert
+	ko apply -f config;
+	ko apply -f config/interceptors;
+
+.PHONY:  clean_tekton
+clean_tekton: | ; $(info $(M) deleteing tekton from local cluster …) @ ## Deleteing tekton on local clustert
+	-ko delete -f config;
+	-ko delete -f config/interceptors;
```
Comment (savitaashture) [inline Makefile:210]: It should be reverse way i mean

```suggestion
	-ko delete -f config/interceptors;
	-ko delete -f config;
```

File: pkg/apis/triggers/v1alpha1/interceptor_types.go [hunk 2]
```diff
@@ -110,6 +110,9 @@ type InterceptorRequest struct {
 	// Header are the headers for the incoming HTTP event
 	Header map[string][]string `json:"header,omitempty"`
 
+	// Form is the form data for the incoming HTTP event
+	Form map[string][]string `json:"form,omitempty"`
```
Comment (savitaashture) [inline pkg/apis/triggers/v1alpha1/interceptor_types.go:114]: Here Form is of type `map[string][]string` but in v1beta1/interceptor_types.go its `string`
Is it intentional :thinking:

File: pkg/sink/sink.go [hunk 3]
```diff
@@ -431,6 +435,10 @@ func (r Sink) processTrigger(t triggersv1.Trigger, el *triggersv1.EventListener,
 	if iresp != nil && iresp.Extensions != nil {
 		extensions = iresp.Extensions
 	}
+
+	log.Infof("*** show extensions %w:", extensions)
+	log.Infof("*** show finalPayload %w:", finalPayload)
```
Comment (savitaashture) [inline pkg/sink/sink.go:440]:
```suggestion

```

File: pkg/interceptors/interceptors.go [hunk 4]
```diff
@@ -115,7 +115,9 @@ func ResolveToURL(getter InterceptorGetter, name string) (*apis.URL, error) {
 }
 
 func Execute(ctx context.Context, client *http.Client, req *triggersv1beta1.InterceptorRequest, url string) (*triggersv1beta1.InterceptorResponse, error) {
+
```
Comment (savitaashture) [inline pkg/interceptors/interceptors.go:118]:
```suggestion
```

File: pkg/interceptors/interceptors.go [hunk 5]
```diff
@@ -115,7 +115,9 @@ func ResolveToURL(getter InterceptorGetter, name string) (*apis.URL, error) {
...
 	b, err := json.Marshal(req)
+
```
Comment (savitaashture) [inline pkg/interceptors/interceptors.go:120]:
```suggestion
```

File: pkg/sink/sink.go [hunk 6]
```diff
@@ -110,6 +113,7 @@ func (r Sink) emitEvents(recorder record.EventRecorder, el *triggersv1.EventList
 
 // HandleEvent processes an incoming HTTP event for the event listener.
 func (r Sink) HandleEvent(response http.ResponseWriter, request *http.Request) {
+
```
Comment (savitaashture) [inline pkg/sink/sink.go:116]:
```suggestion
```

File: pkg/apis/triggers/v1alpha1/interceptor_types.go (hunk 2 above)
Comment (ilan-pinto) [reply pkg/apis/triggers/v1alpha1/interceptor_types.go:114]: no. ill fix it.

**Comment by ilan-pinto** [comment]
Put this PR on hold 
I have decided to take a different approach to solve this issue 
the new branch is under can be seen in PR #1548

**Comment by khrm** [comment]
@ilan-pinto I think this pr isn't needed. Closing it.

**Pull request closed without merging**